permissionsdispatcher-processor/src/main/java/permissions/dispatcher/processor/JavaFileBuilder.java

```diff
@@ -74,12 +74,26 @@ private static MethodSpec createOnRequestPermissionsResult(ClassName target, Lis
                 .addParameter(int[].class, "grantResults")
                 .returns(void.class)
                 .beginControlFlow("switch (requestCode)");
-        for (ExecutableElement method : methods) {
+        for (ExecutableElement method : permissionMethods) {
             String methodName = method.getSimpleName().toString();
             methodBuilder
                     .addCode("case $N:\n", getRequestCodeFieldName(methodName))
                     .beginControlFlow("if ($T.verifyPermissions(grantResults))", PERMISSION_UTILS)
-                    .addStatement("target.$N()", methodName)
+                    .addStatement("target.$N()", methodName);
+            NeedsPermission needsPermission = method.getAnnotation(NeedsPermission.class);
+            String needsPermissionValue = needsPermission != null ? needsPermission.value() : "";
+            ExecutableElement deniedPermission = clazz.getDeniedPermissionFromValue(needsPermissionValue);
```

Comment:
I'd like to pull out codes which try to get 'deniedPermission' to single static method like 'getDeniedPermission' or like that. Because it simplifies a logic flow.